### ISSUE TITLE
Add the Smoke test PR check for Dashboard repository

### ISSUE THREAD

**Issue by dmytro-ndp** [eclipse/che#22045]
### Is your task related to a problem? Please describe

Run Smoke GUI test on new pull request or new commit to pull request in repo https://github.com/eclipse-che/che-dashboard

### Describe the solution you'd like

Create GitHub actions on pull request events to do:
1. Wait on successful build of GitHub action "[docker-build](https://github.com/eclipse-che/che-dashboard/blob/main/.github/workflows/pr.yml#LL70C12-L70C12)" [e.g. https://github.com/eclipse-che/che-dashboard/actions/runs/4952254169/jobs/8858567817)
2. [Deploy OpenShift instance on Prow CI](https://github.com/eclipse-che/che-dashboard/blob/main/.ci/oci-dashboard-happy-path.sh).
3. Install Eclipse Che next using dashboard image from PR by patching CheCluster [1], e.g.:
```
apiVersion: org.eclipse.che/v2
spec:
  components:
    dashboard:
      dashboardImage: '$CI_CHE_DASHBOARD_IMAGE'
```

CI_CHE_DASHBOARD_IMAGE = quay.io/eclipse-che/che-dashboard:pr-<PR_NUMBER>

4. Run smoke GUI test against Eclipse Che instance.
5. Collect the logs and shutdown OpenShift instance.

[1] https://github.com/eclipse-che/che-dashboard/blob/main/.ci/oci-dashboard-happy-path.sh#L24-L29

Technical debts:
1) Get rid of "test-all-devfiles" command: https://github.com/eclipse/che/blob/main/tests/e2e/package.json#L13
2) Replace redundant outdated constant `TS_SELENIUM_USERNAME` with `TS_SELENIUM_OCP_USERNAME`
3) Replace redundant outdated constant `TS_SELENIUM_PASSWORD` with `TS_SELENIUM_OCP_PASSWORD`
